hudi-cli/src/main/java/org/apache/hudi/cli/commands/CompactionCommand.java

```diff
@@ -71,10 +72,12 @@
 
   private static final String TMP_DIR = "/tmp/";
 
-  @CliAvailabilityIndicator({"compactions show all", "compaction show", "compaction run", "compaction schedule"})
-  public boolean isAvailable() {
-    return (HoodieCLI.tableMetadata != null)
-        && (HoodieCLI.tableMetadata.getTableType() == HoodieTableType.MERGE_ON_READ);
+  private HoodieTableMetaClient checkAndGetMetaClient() {
+    HoodieTableMetaClient client = HoodieCLI.getTableMetaClient();
+    if (client.getTableType() != HoodieTableType.MERGE_ON_READ) {
+      throw new InvalidTableTypeException("Compactions can only be run for table type : MERGE_ON_READ");
```

Comment:
How about using `HoodieException` here and avoid creating new InvalidTableTypeException class since it is only used here currently?

Reply:
> How about using `HoodieException` here and avoid creating new InvalidTableTypeException class since it is only used here currently?

Yes, changed to HoodieException and I will delete InvalidTableTypeException.